python/cudf/cudf/tests/test_decimal.py

```diff
@@ -151,14 +194,15 @@ def test_typecast_to_from_decimal(data, from_dtype, to_dtype):
 )
 @pytest.mark.parametrize(
     "from_dtype",
-    [Decimal64Dtype(7, 2), Decimal64Dtype(11, 4), Decimal64Dtype(18, 10)],
+    [Decimal64Dtype(7, 2), Decimal64Dtype(11, 4), Decimal64Dtype(17, 10)],
 )
-@pytest.mark.parametrize("to_dtype", FLOAT_TYPES)
+@pytest.mark.parametrize("to_dtype", NUMERIC_TYPES)
 def test_typecast_from_decimal(data, from_dtype, to_dtype):
-    actual = data.astype(from_dtype)
-    pa_arr = actual.to_arrow().cast(to_dtype, safe=False)
+    got = data.astype(from_dtype)
+    pa_arr = got.to_arrow().cast(to_dtype, safe=False)
 
-    actual = actual.astype(to_dtype)
+    got = got.astype(to_dtype)
     expected = cudf.Series(NumericalColumn.from_arrow(pa_arr))
 
-    assert_eq(actual, expected)
+    assert_eq(got, expected)
+    assert_eq(got.dtype, expected.dtype)
```

Comment:
```suggestion
    assert_eq(got, expected, check_dtype=True)
```

You can validate the dtype & series in a single go like this. Similarly in other tests too..

Reply:
So this issue is that even with `check_dtype=True` the `assert_eq` function doesn't check if precision and scale are equal, but `assert_eq(got.dtype, expected.dtype)` does.

Reply:
Ah, that is because doesn't have a decimal type yet..Then we should keep this for now.